tests/test_realization.py

```diff
@@ -815,3 +816,88 @@ def test_drop():
 
     real.drop("parameters")
     assert "parameters.txt" not in real.keys()
+
+
+def test_find_files_comps():
+    """Test the more exotic features of find_files
+
+    Components extracted from filenames.
+    """
+
+    testdir = os.path.dirname(os.path.abspath(__file__))
+    realdir = os.path.join(testdir, "data/testensemble-reek001", "realization-0/iter-0")
+    real = ensemble.ScratchRealization(realdir)
+
+    # Make some filenames we can later "find", including some problematic ones.
+    findable_files = [
+        "foo--bar--com.gri",
+        "foo-bar--com.gri",
+        "foo---bar--com.gri",
+        "--bar--.gri",
+    ]
+    for filename in findable_files:
+        with open(os.path.join(realdir, filename), "w") as fileh:
+            fileh.write("baah")
+
+    real.find_files("*.gri")
+
+    files_df = real.files.set_index("BASENAME")
+    assert "COMP0" not in real.files  # We are 1-based, not zero.
+    assert "COMP1" in real.files
+    assert "COMP2" in real.files
+    assert "COMP3" in real.files
+    assert "COMP4" not in real.files
+
+    assert files_df.loc["foo--bar--com.gri"]["COMP1"] == "foo"
+    assert files_df.loc["foo--bar--com.gri"]["COMP2"] == "bar"
+    assert files_df.loc["foo--bar--com.gri"]["COMP3"] == "com"
+    assert files_df.loc["foo-bar--com.gri"]["COMP1"] == "foo-bar"
+    assert files_df.loc["foo-bar--com.gri"]["COMP2"] == "com"
+    assert files_df.loc["foo---bar--com.gri"]["COMP1"] == "foo"
+    assert files_df.loc["foo---bar--com.gri"]["COMP2"] == "-bar"
+    assert files_df.loc["foo---bar--com.gri"]["COMP3"] == "com"
+    assert files_df.loc["--bar--.gri"]["COMP1"] == ""
+    assert files_df.loc["--bar--.gri"]["COMP2"] == "bar"
+    assert files_df.loc["--bar--.gri"]["COMP3"] == ""
+
+    # Cleanup
+    for filename in findable_files:
+        if os.path.exists(os.path.join(realdir, filename)):
+            os.unlink(os.path.join(realdir, filename))
+
+
+def test_find_files_yml():
+    """Test the more exotic features of find_files
+
+    Meta-data in yaml files.
+    """
+    testdir = os.path.dirname(os.path.abspath(__file__))
+    realdir = os.path.join(testdir, "data/testensemble-reek001", "realization-0/iter-0")
+    real = ensemble.ScratchRealization(realdir)
+
+    # Setup example files with some yaml data:
+    findable_files = ["grid1.gri", "grid2.gri"]
+    for filename in findable_files:
+        with open(os.path.join(realdir, filename), "w") as fileh:
+            fileh.write("baah")
+        yamlfile = "." + filename + ".yml"
+        with open(os.path.join(realdir, yamlfile), "w") as fileh:
+            fileh.write(yaml.dump(dict(a=dict(x=1, y=2), b="bar")))
```

Comment:
Could you give an example of the use case of having a yaml files with key-values pairs? Not sure from the test how this yaml file would relates to the *gri files?

Reply:
F.ex. *.gri files (surfaces) can have an arbitrary amount of meta-information attached to them, hints for visualization, coordinates system, dates, source, etc etc. 

The functionality for  flattening yaml-dict structures to key-value is in order to include the values as tabular data when you ask an ensemble object for its `.files`. An application can have its own selection logic based on the meta-data for which surfaces to display and how.

Reply:
See https://github.com/equinor/fmu-ensemble/blob/rmrc/tests/test_rmrc.py#L48 for example of usage.

Reply:
Make sense and the another implementation is if the meta-information is directly in the filename denoted by --